daemon/daemon.go

```diff
@@ -196,16 +196,15 @@ func (daemon *Daemon) RegistryHosts(host string) ([]docker.RegistryHost, error)
 	}
 	conf := daemon.registryService.ServiceConfig().IndexConfigs
 	for k, v := range conf {
-		c := resolverconfig.RegistryConfig{}
+		c := m[k]
 		if !v.Secure {
 			t := true
 			c.PlainHTTP = &t
 			c.Insecure = &t
 		}
 		m[k] = c
 	}
-	if _, ok := m[host]; !ok && daemon.registryService.IsInsecureRegistry(host) {
-		c := resolverconfig.RegistryConfig{}
+	if c, ok := m[host]; !ok && daemon.registryService.IsInsecureRegistry(host) {
```

Comment:
Not something new in this PR, but `IsInsecureRegistry` _also_ appears to be checking `config.IndexConfigs`. Are we doing the same thing _twice_ in this function?

Reply:
Hmm, good question, I thought it was needed because it also does the `isCIDRMatch` check for insecure registries specified by CIDR (`InsecureRegistryCIDRs`).
But if I understand correctly, it actually doesn't handle them, because these are not a part of `IndexConfigs` anyway...

Reply:
Yeah, the overall combination of mirrors and insecure registries (and how they're both "separate" and "depending on each other") is ... messy